cmd/company/main.go

```diff
@@ -19,22 +19,34 @@ func main() {
 
 	// "postgresql://user:password@host:port/dbname"
 	connString := `postgresql://appuser:appuser@127.0.0.1:5432/app_db`
-	DB, err := postgres.NewDB(ctx, connString)
+	db, err := postgres.NewDB(ctx, connString)
 	if err != nil {
 		log.Fatal(err)
 	}
+	defer db.Close()
 
-	repo := storage.NewDB(DB)
+	repo := storage.NewDB(db)
 	a := starter.NewApp(repo)
 	h := handler.NewRouter(repo)
 	srv := server.NewServer(":8000", h)
 
 	wg := &sync.WaitGroup{}
-	wg.Add(1)
-
-	go a.Serve(ctx, wg, srv)
-
-	<-ctx.Done()
-	cancel()
-	wg.Wait()
+	go func() {
+		wg.Add(1)
+		a.Serve(ctx, srv)
+		wg.Done()
+	}()
+
+	for {
+		select {
+		case err := <-srv.Err:
+			cancel()
+			wg.Wait()
+
+			log.Fatal(err)
+		case <-ctx.Done():
+			cancel()
+			wg.Wait()
+		}
```

Comment:
1. Поскольку и `err`, и `ctx.Done()` обрабатываются одинаково, можно вынести обработку за `select`.

2. Логировать с уровнем `Fatal` здесь плохо, т.к. `Fatal` приведет к немедленному завершению приложения (и даже `defer`'ы не выполнятся). В этом месте приложения мы уже и так выходим, уровня `Error` достаточно (сами логи важны, чтобы в них потом покопаться, да).

```suggestion
		case err := <-srv.Err:
			log.Errorf("got an error from the server: %w", err)
		case <-ctx.Done():
		}
		
		cancel()
		wg.Wait()
```